hoomd/md/EvaluatorPairExpandedLJ.h

```diff
@@ -0,0 +1,224 @@
+          lj2(_params.epsilon_x_4 * _params.sigma_6), delta(_params.delta)
+        {
+        }
+
+    //! ExpandedLJ does not use diameter
+    DEVICE static bool needsDiameter()
+        {
+        return false;
+        }
+    //! Accept the optional diameter values
+    /*! \param di Diameter of particle i
+        \param dj Diameter of particle j
+    */
+    DEVICE void setDiameter(Scalar di, Scalar dj) const { }
+
+    //! ExpandedLJ does not use charge
+    DEVICE static bool needsCharge()
+        {
+        return false;
+        }
+    //! Accept the optional diameter values
+    /*! \param qi Charge of particle i
+        \param qj Charge of particle j
+    */
+    DEVICE void setCharge(Scalar qi, Scalar qj) const { }
+
+    //! Evaluate the force and energy
+    /*! \param force_divr Output parameter to write the computed force divided by r.
+        \param pair_eng Output parameter to write the computed pair energy
+        \param energy_shift If true, the potential must be shifted so that V(r) is continuous at the
+       cutoff \note There is no need to check if rsq < rcutsq in this method. Cutoff tests are
+       performed in PotentialPair.
+
+        \return True if they are evaluated or false if they are not because we are beyond the cutoff
+    */
+    DEVICE bool evalForceAndEnergy(Scalar& force_divr, Scalar& pair_eng, bool energy_shift)
+        {
+        // precompute some quantities
+        Scalar rinv = fast::rsqrt(rsq);
+        Scalar r = Scalar(1.0) / rinv;
+        Scalar rcutinv = fast::rsqrt(rcutsq);
+        Scalar rcut = Scalar(1.0) / rcutinv;
+
+        // compute the force divided by r in force_divr
+        if (r < rcut && lj1 != 0)
+            {
+            Scalar rmd = r - delta;
+            Scalar rmdinv = Scalar(1.0) / rmd;
+            Scalar rmd2inv = rmdinv * rmdinv;
+            Scalar rmd6inv = rmd2inv * rmd2inv * rmd2inv;
+            force_divr
+                = rinv * rmdinv * rmd6inv * (Scalar(12.0) * lj1 * rmd6inv - Scalar(6.0) * lj2);
+
+            pair_eng = rmd6inv * (lj1 * rmd6inv - lj2);
+
+            if (energy_shift)
+                {
+                Scalar rcut2inv = rcutinv * rcutinv;
+                Scalar rcut6inv = rcut2inv * rcut2inv * rcut2inv;
+                pair_eng -= rcut6inv * (lj1 * rcut6inv - lj2);
+                }
+            return true;
+            }
+        else
+            return false;
+        }
+
+#ifndef __HIPCC__
+    //! Get the name of this potential
+    /*! \returns The potential name. Must be short and all lowercase, as this is the name energies
+       will be logged as via analyze.log.
+     */
+    static std::string getName()
+        {
+        return std::string("expanded_lj");
+        }
+
+    std::string getShapeSpec() const
+        {
+        throw std::runtime_error("Shape definition not supported for this pair potential.");
+        }
+#endif
+
+    protected:
```

Comment:
Why protected here instead of private? If there aren't any subclasses that use these data members, this should be private.

Reply:
offline, Josh said to keep protected for consistency with the rest of the code